internal/appsec/dyngo/instrumentation/common.go

```diff
@@ -65,6 +65,13 @@ func (s *SecurityEventsHolder) Events() []json.RawMessage {
 	return s.events
 }
 
+// ClearEvents clears the list of stored events
+func (s *SecurityEventsHolder) ClearEvents() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.events = []json.RawMessage{}
```

Comment:
Keep the underlying memory as long as you keep the operation:
```suggestion
	s.events = s.events[0:0]
```